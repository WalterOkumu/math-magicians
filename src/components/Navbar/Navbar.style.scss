nav {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding-top: 10px;

  .logo {
    font-size: 50px;
    font-weight: bold;
    padding-left: 20px;
  }

  .navbar {
    display: flex;
    flex-direction: row;
    align-items: center;
    text-decoration: underline;
    gap: 20px;
    padding-right: 20px;

    .navbar-item {
      font-weight: bold;
    }
  }
}
