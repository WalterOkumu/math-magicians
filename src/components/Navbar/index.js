import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navbar.style.scss';

const Navbar = () => (
  <nav>
    <div className="logo">
      Math Magicians
    </div>
    <ul className="navbar">
      <li className="navbar-item">
        <NavLink to="/">
          Home
        </NavLink>
      </li>
      <li>
        &#124;
      </li>
      <li className="navbar-item">
        <NavLink to="/calculator">
          Calculator
        </NavLink>
      </li>
      <li>
        &#124;
      </li>
      <li className="navbar-item">
        <NavLink to="/quote">
          Quote
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navbar;
