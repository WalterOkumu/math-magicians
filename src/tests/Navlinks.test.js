import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Testing Navbar Links', () => {
  test('Testing that all links are present', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );

    expect(screen.getAllByText('Home')).toMatchSnapshot();
    expect(screen.getAllByText('Calculator')).toMatchSnapshot();
    expect(screen.getAllByText('Quote')).toMatchSnapshot();
  });

  test('Test navigate to Home Page', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );

    fireEvent.click(screen.getByText('Home'));

    expect(screen.getByText('Home')).toMatchInlineSnapshot(`
<a
  aria-current="page"
  class="active"
  href="/"
>
  Home
</a>
`);
  });

  test('Test navigate to Calculator Page', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );

    fireEvent.click(screen.getByText('Calculator'));

    expect(screen.getByText('Calculator')).toMatchInlineSnapshot(`
<a
  aria-current="page"
  class="active"
  href="/calculator"
>
  Calculator
</a>
`);
  });

  test('Test navigate to Quote Page', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );

    fireEvent.click(screen.getByText('Quote'));

    expect(screen.getByText('Quote')).toMatchInlineSnapshot(`
<a
  aria-current="page"
  class="active"
  href="/quote"
>
  Quote
</a>
`);
  });
});
