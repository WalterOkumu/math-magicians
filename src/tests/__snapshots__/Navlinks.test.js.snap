// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar Links Testing that all links are present 1`] = `
Array [
  <a
    aria-current="page"
    class="active"
    href="/"
  >
    Home
  </a>,
]
`;

exports[`Testing Navbar Links Testing that all links are present 2`] = `
Array [
  <a
    class=""
    href="/calculator"
  >
    Calculator
  </a>,
]
`;

exports[`Testing Navbar Links Testing that all links are present 3`] = `
Array [
  <a
    class=""
    href="/quote"
  >
    Quote
  </a>,
]
`;
